### PR TITLE
fix(ivy): make ViewRef.detectChanges work with embedded views (FW-749)

### PR DESCRIPTION
**Pull request by AndrewKushnir**
In `ViewRef.detectChanges`, we are passing `ViewRef.context` into `detectChanges` to trigger change detection. This only makes sense for component `ViewRefs` (i.e. injected `ChangeDetectorRefs`) because with embedded views, `context` is not a component instance where the view has been monkey-patched. It's a just a normal object, so the view will be undefined.

In order to resolve this problem, we now invoke `detectChangesInternal` and also pass `LView` (to make sure we always have a view available).

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/core/test/render3/change_detection_spec.ts [hunk 1]
```diff
@@ -145,10 +145,13 @@ describe('change detection', () => {
       fixture.update();
       expect(fixture.html).toEqual('<structural-comp>one</structural-comp>');
 
-      structuralComp.create();
+      const viewRef: EmbeddedViewRef<any> = structuralComp.create();
       fixture.update();
       expect(fixture.html).toEqual('<structural-comp>one</structural-comp>Temp content');
 
+      viewRef.detectChanges();
+      expect(fixture.html).toEqual('<structural-comp>one</structural-comp>Temp content');
```
Comment (kara) [inline packages/core/test/render3/change_detection_spec.ts]: This doesn't really test that we are successfully running change detection on the correct view; it just tests that the DOM hasn't changed. 

Can we add a binding into the dynamic template and check to see that it was updated properly (and that change detection does *not* run in the parent view)? You will probably want to pass in a context object when creating the embedded view to achieve this.

**Review by kara — CHANGES_REQUESTED** [review]
Change looks good, but we need a stronger test

File: packages/core/test/render3/change_detection_spec.ts (hunk 1 above)
Comment (AndrewKushnir) [reply packages/core/test/render3/change_detection_spec.ts]: Thanks Kara. I've updated the test to include binding into embedded view to make sure change detection is properly isolated.

File: packages/core/test/render3/change_detection_spec.ts [hunk 2]
```diff
@@ -91,13 +91,23 @@ describe('change detection', () => {
     it('should support detectChanges on components that have LContainers', () => {
       let structuralComp !: StructuralComp;
 
+      function FooTemplate(rf: RenderFlags, ctx: any) {
+        debugger;
```
Comment (pkozlowski-opensource) [inline packages/core/test/render3/change_detection_spec.ts]: `debugger`; left

Comment (AndrewKushnir) [reply packages/core/test/render3/change_detection_spec.ts]: Ah! Thanks @pkozlowski-opensource, removed.

File: packages/core/test/render3/change_detection_spec.ts [hunk 3]
```diff
@@ -145,13 +139,19 @@ describe('change detection', () => {
...
+      expect(fixture.html).toEqual('<structural-comp>one</structural-comp>one');
 
+      // check embedded view update
       structuralComp.value = 'two';
+      viewRef.detectChanges();
+      expect(fixture.html).toEqual('<structural-comp>one</structural-comp>two');
+
+      // check root view update
+      structuralComp.value = 'three';
       detectChanges(structuralComp);
```
Comment (kara) [inline packages/core/test/render3/change_detection_spec.ts]: Maybe use `fixture.update()` here? Triggering change detection at this level means that the dynamic view won't be updated.

**Review by kara — APPROVED** [review]
LGTM, merge when ready

File: packages/core/test/render3/change_detection_spec.ts (hunk 3 above)
Comment (AndrewKushnir) [reply packages/core/test/render3/change_detection_spec.ts]: Thanks Kara, I've updated the test.

**Comment by kara** [comment]
[presubmit](http://test/OCL:224808782:BASE:224808991:1544455926294:bb32247d
)

**Comment by AndrewKushnir** [comment]
[Presubmit #2 (due to flaky test)](https://test.corp.google.com/ui#id=OCL:224808782:BASE:224847949:1544469511979:dc250831)

**Pull request closed without merging**